lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -39,14 +40,27 @@ export class AuthCache extends BrowserStorage {// Singleton
         const errorDescKey = `${Constants.cachePrefix}.${ErrorCacheKeys.ERROR_DESC}`;
 
         const idTokenValue = super.getItem(idTokenKey);
-        const clientInfoValue = super.getItem(clientInfoKey);
-        const errorValue = super.getItem(errorKey);
-        const errorDescValue = super.getItem(errorDescKey);
 
-        const values = [idTokenValue, clientInfoValue, errorValue, errorDescValue];
-        const keysToMigrate = [PersistentCacheKeys.IDTOKEN, PersistentCacheKeys.CLIENT_INFO, ErrorCacheKeys.ERROR, ErrorCacheKeys.ERROR_DESC];
+        let idToken;
 
-        keysToMigrate.forEach((cacheKey, index) => this.duplicateCacheEntry(cacheKey, values[index], storeAuthStateInCookie));
+        if (idTokenValue) {
+            try {
+                idToken = new IdToken(idTokenValue);
```

Comment:
What would cause this to throw?

Reply:
Potentially if the raw idToken string cannot be decoded and/or JSON parsed